platforms/vitis/cl_firesim/Makefile

```diff
@@ -118,18 +135,26 @@ ipgen_scripts = $(shell find design/ -iname "*.ipgen.tcl")
 	# Move into the simulation or bitstream-build directory
 	cd $(@D)/..; $(VIVADO) -mode batch -source $(buildroot)/scripts/gen_xo.tcl -tclargs $@ $(PROJECT_NAME) $(TARGET) $(DEVICE) $(XSA)
 
+build_strategy_path = $(STRATEGIES_DIR)/strategy_$(STRATEGY).cfg
+
+ifneq ($(shell test -f $(build_strategy_path); echo $$?),0)
+$(error Build strategy config at $(build_strategy_path) does not exist)
+endif
+VPP_CONFIG_OVERLAYS += --config $(build_strategy_path)
+
 # link and package the xclbin
-%/$(PROJECT_NAME).link.xclbin: %/$(PROJECT_NAME).xo
+%/$(PROJECT_NAME).link.xclbin: %/$(PROJECT_NAME).xo $(build_strategy_path)
 	mkdir -p $(@D)
 	# Move into temp directory to avoid polluting top-level workdir
-	cd $(@D)/..; $(VPP) $(VPP_TARGET_SPECIFIC_ARGS) $(VPP_COMMON_ARGS) \
+	cd $(@D)/..; $(VPP) $(VPP_CONFIG_OVERLAYS) $(VPP_TARGET_SPECIFIC_ARGS) $(VPP_COMMON_ARGS) \
```

Comment:
Shouldn't the default `VPP_TARGET_SPECIFIC_ARGS` be 1st so that the `build-bitstream.cfg` can be read in 1st?

Reply:
I wanted the overlays to be the highest precedence of the config files. The strategies should be non-overlapping with build-bitstream.cfg, but i want whatever to be passed in via the CONFIG_OVERLAY to win over everything except explicit commandline arguments